client/app/widgets/Comments/__tests__/Comments.spec.jsx

```diff
@@ -83,41 +80,33 @@ const component = <Comments formProps={formProps} />;
 
 describe('Comments', () => {
   beforeEach(() => {
-    axiosPostSpy = jest.spyOn(axios, 'post').mockImplementation(() => Promise.resolve({
+    jest.spyOn(axios, 'post').mockResolvedValue({
       data: { comment },
-    }));
-    axiosDeleteSpy = jest.spyOn(axios, 'delete').mockImplementation(() => Promise.resolve({
+    });
+    jest.spyOn(axios, 'delete').mockResolvedValue({
       data: { id },
-    }));
+    });
   });
 
   it('renders correctly', () => {
-    let wrapper = null;
-    expect(() => {
-      wrapper = mount(component);
-    }).not.toThrow();
-    expect(wrapper).not.toBeNull();
+    render(component);
+    expect(screen.queryByRole('article')).not.toBeInTheDocument();
+    expect(screen.getByRole('button', { name: 'Submit' })).toBeInTheDocument();
   });
 
   it('add and delete a comment', async () => {
-    const wrapper = mount(component);
-    expect(wrapper.find('.comment').exists()).toEqual(false);
-    act(() => {
-      wrapper
-        .find('input[name="comment[comment]"]')
-        .simulate('change', { currentTarget: { value } });
+    render(component);
+    expect(screen.queryByRole('article')).not.toBeInTheDocument();
 
-      wrapper.find('select#comment_visibility').prop('onChange')({
-        currentTarget: { value: 'private' },
-      });
-    });
-    wrapper.find('input[type="submit"]').simulate('click');
-    await axiosPostSpy();
-    wrapper.update();
-    expect(wrapper.find('.comment').exists()).toEqual(true);
-    wrapper.find('.storyActionsDelete').find('a').simulate('click');
-    await axiosDeleteSpy();
-    wrapper.update();
-    expect(wrapper.find('.comment').exists()).toEqual(false);
+    userEvent.type(screen.getByRole('textbox'));
+    userEvent.selectOptions(screen.getByRole('combobox'), 'private');
+    userEvent.click(screen.getByRole('button', { name: 'Submit' }));
+
+    await waitFor(() => expect(screen.queryByRole('article')).toBeInTheDocument());
+    expect(screen.queryByRole('article')).toHaveTextContent('Hey');
```

Comment:
Nice 👍 I like this!

Minor note: when querying for presence, `getByRole` [can be used](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-query-variants-for-anything-except-checking-for-non-existence) instead.

Reply:
I updated to use `getByRole`.